slasher/db/kv/spanner.go

```diff
@@ -0,0 +1,282 @@
+package kv
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/boltdb/bolt"
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+	"github.com/prysmaticlabs/prysm/slasher/detection/attestations/types"
+	log "github.com/sirupsen/logrus"
+	"go.opencensus.io/trace"
+)
+
+var highestEpoch uint64
+
+func saveToDB(db *Store) func(uint64, uint64, interface{}, int64) {
+	// Returning the function here so we can access the DB properly from the OnEvict.
+	return func(epoch uint64, _ uint64, value interface{}, cost int64) {
+		log.Tracef("evicting span map for epoch: %d", epoch)
+		err := db.update(func(tx *bolt.Tx) error {
+			bucket := tx.Bucket(validatorsMinMaxSpanBucket)
+			epochBucket, err := bucket.CreateBucketIfNotExists(bytesutil.Bytes8(epoch))
+			if err != nil {
+				return err
+			}
+			spanMap := value.(map[uint64]types.Span)
```

Comment:
needs an ok check for safety `spanMap, ok := value.(map[uint64]types.Span)` and check !ok

Reply:
thanks